### ISSUE TITLE
[BUG] False Positive for `the MatchError matcher used to assert a non error type` in `Eventually` cases with `SpecContext` arguments

### ISSUE THREAD

**Issue by timebertt** [nunnatsa/ginkgolinter#190]
**Describe the bug**

The linter raises false positives for the correct usage of `MatchError` on `Eventually` statements with a `SpecContext` argument:

```go
It("should work", func(ctx SpecContext) {
	Eventually(ctx, func() error {
		return fmt.Errorf("foo")
	}).Should(MatchError("foo"))
}, SpecTimeout(time.Second))
```

```
ginkgo-linter: the MatchError matcher used to assert a non error type (ctx)
```

This pattern is the recommended way for implementing interruptible specs when using `Eventually` ([docs](https://onsi.github.io/ginkgo/#using-speccontext-with-gomegas-eventually)).

**To Reproduce**

```
package main_test

import (
	"fmt"
	"testing"
	"time"

	. "github.com/onsi/ginkgo/v2"
	. "github.com/onsi/gomega"
)

func TestE2E(t *testing.T) {
	RegisterFailHandler(Fail)
	RunSpecs(t, "demo")
}

var _ = Describe("the MatchError matcher used to assert a non error type", func() {
	It("should work", func(ctx SpecContext) {
		Eventually(func() error {
			return fmt.Errorf("foo")
		}).WithContext(ctx).Should(MatchError("foo"))
	}, SpecTimeout(time.Second))
})
```

**Expected behavior**

No linter error

**Environment:**
 - OS: macOS
 - Version: v0.19.0

**Additional context**

Note that passing the context as the first `Eventually` arg is convenience over passing it to `WithContext`.
I.e., this is semantically equivalent but does not raise a false positive:

```go
It("should work", func(ctx SpecContext) {
	Eventually(func() error {
		return fmt.Errorf("foo")
	}).WithContext(ctx).Should(MatchError("foo"))
}, SpecTimeout(time.Second))
```

**Comment by nunnatsa** [comment]
@timebertt , Just released version [v0.19.1](https://github.com/nunnatsa/ginkgolinter/releases/tag/v0.19.1) with a fix.

**Comment by timebertt** [comment]
@nunnatsa that was fast, thank you very much! 🚀

**Comment by nunnatsa** [comment]
> [@nunnatsa](https://github.com/nunnatsa) that was fast, thank you very much! 🚀

:blush:

**Comment by nunnatsa** [comment]
golangci-lint [v1.64.6](https://github.com/golangci/golangci-lint/releases/tag/v1.64.6) was released with this fix
